src/main/java/frc/robot/subsystems/drive/Drive.java

```diff
@@ -389,6 +428,149 @@ public Command getDriveToPoseCommand() {
     return AutoBuilder.pathfindToPose(targetPose, constraints, 0.0);
   }
 
+  /**
+   * checks if driver station alliance is red
+   *
+   * @return true if alliance is red
+   */
+  public boolean isAllianceRed() {
+    return DriverStation.getAlliance().orElse(Alliance.Blue).equals(Alliance.Red);
+  }
+
+  /**
+   * gets tag to use for final alignment with vision
+   *
+   * @return int representing tag id to use
+   */
+  public int getTagIdForReef() {
+    boolean allianceRed = this.isAllianceRed();
+    switch (desiredLocation) {
```

Comment:
Same comment as other switch statement, you can avoid repeating the same return statement by just stacking cases that are the same